src/commands.ts

```diff
@@ -59,45 +100,60 @@ async function buildCurrentProject(args, dotNetSdk: DotNetInfo, commandObserver:
 		project = args.fsPath;
 	}
 
-	await buildProjects(dotNetSdk, [project], commandObserver, cancelToken).then(() => {
-		commandObserver.buildInProgress = false;
-	});
+	await buildProjects(dotNetSdk, [project], commandObserver, cancelToken);
 }
 
-async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
 	let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
 	await buildProjects(dotNetSdk, projects.map(p => p.fsPath), commandObserver, cancelToken);
 }
 
-async function buildProjects(dotNetSdk: DotNetInfo, projects: string[], commandObserver: CommandObserver, cancelToken: vscode.CancellationToken) {
+async function buildProjects(dotNetSdk: DotNetInfo, projects: string[], commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<BuildResult[]> {
+	var buildResult = new Array<BuildResult>();
 	if (commandObserver.buildInProgress) {
 		vscode.window.showErrorMessage(localize('extension.existingBuildInProgressMessage', 'There is a build already running, please cancel the build before starting a new one'));
-		return;
+		return Promise.resolve(buildResult);
 	}
-	commandObserver.buildInProgress = true;
+
 	try {
+		var successProjectCount = 0;
+		var failedProjectCount = 0;
+		commandObserver.buildInProgress = true;
 		commandObserver.resetOutputChannel();
+		vscode.workspace.saveAll();
 		let unsupportedProjects = await validateProjectSDK(projects, commandObserver);
 		if (unsupportedProjects.length > 0) {
 			projects = projects.filter(p => unsupportedProjects.indexOf(p) < 0);
+			unsupportedProjects.map(p => buildResult.push({ project: p, status: buildStatus.Skipped }));
 		}
+
 		for (let project of projects) {
-			if (cancelToken.isCancellationRequested) {
+			if (cancelToken && cancelToken.isCancellationRequested) {
 				return;
 			}
-			await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
+			await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken).then(() => {
+				successProjectCount++;
+				commandObserver.logToOutputChannel(localize('extension.buildEndMessage', 'Done building project {0}\n', project));
+				buildResult.push({ project: project, status: buildStatus.Success })
+			}, () => {
+				failedProjectCount++;
+				commandObserver.logToOutputChannel(localize('extension.buildFailMessage', 'Done building project {0} -- FAILED\n', project));
+				buildResult.push({ project: project, status: buildStatus.Failure })
+			});
 		}
+		commandObserver.logToOutputChannel(localize('extension.buildSummaryMessage', '======== Build: {0} succeeses or up-to-date, {1} failed, {2} skipped ========', successProjectCount, failedProjectCount, unsupportedProjects.length));
```

Comment:
Isn't all this produced by msbuild itself? Why do you need to generate these strings yourself?
This makes no sense to me, to be honest. If you build a proj file, that internally references your projects + something else - you don't want to print this just for your projects, msbuild will do it for all projects it builds at the very end.

Reply:
msbuild does not give you a summary, this is similar to what visual studio does. It will show you a summary of number of passed, failed and skipped projects in the end.

Reply:
![image](https://user-images.githubusercontent.com/36710109/71536326-b0036380-28c2-11ea-9343-e81c58e9fd5f.png)